Packages/com.unity.inputsystem/InputSystem/Devices/ScreenKeyboard.cs

```diff
@@ -0,0 +1,298 @@
+    internal struct ScreenKeyboardCallbacks
+    {
+        /// <summary>
+        /// Is fired whenever screen keyboard state changes
+        /// </summary>
+        internal Action<ScreenKeyboardState> stateChanged { set; get; }
+
+        /// <summary>
+        /// Is fired whenever input field text changes.
+        /// It doesn't matter if input field is hidden or not, this callback would still be invoked
+        /// </summary>
+        internal Action<string> inputFieldTextChanged { set; get; }
+
+        /// <summary>
+        /// Is fired whenever input field text selection changes.
+        /// </summary>
+        internal Action<RangeInt> inputFieldSelectionChanged { set; get; }
+    }
+
+    /// <summary>
+    /// ScreenKeyboard base class for platform specific implementation.
+    ///
+    /// Known issues:
+    /// * There's a screen keyboard behavior differences in regards to input and focus loss:
+    ///   - [Android] When screen keyboard is shown, the main Unity window stops receiving input events, when clicking on Unity window, screen keyboard looses focus and closes itself
+    ///     (While it's possible to keep screen keyboard opened on focus loss, currently there's no way to make Unity window keep receiving input events like touch while screen keyboard is shown,
+    ///      this matches the behavior in old TouchScreenKeyboard implementation)
+    ///   - [iOS] When screen keyboard is shown, the main Unity window will continue receiving input events, when clicking on Unity window, screen keyboard will continue to be shown
+    ///   - [WSA] When screen keyboard is shown, the main Unity window will continue receiving input events, when clicking on Unity window, screen keyboard will continue to be shown
+    /// </summary>
+    internal abstract class ScreenKeyboard : IDisposable
+    {
+        protected ScreenKeyboardState m_KeyboardState;
+        protected ScreenKeyboardShowParams m_ShowParams;
+        private ScreenKeyboardCallbacks m_Callbacks;
+
+        /// <summary>
+        /// Returns the state of the keyboard
+        /// </summary>
+        public ScreenKeyboardState state
+        {
+            get => m_KeyboardState;
+        }
+
+        /// <summary>
+        /// Cleanup any platform specific resources
+        /// </summary>
+        public virtual void Dispose()
+        {
+        }
+
+        /// <summary>
+        /// Shows the screen keyboard with default options.
+        /// </summary>
+        public void Show()
+        {
+            Show(new ScreenKeyboardShowParams());
+        }
+
+        /// <summary>
+        /// Show the screen keyboard with customized options.
+        /// </summary>
+        /// <param name="showParams"></param>
+        /// <param name="callbacks"></param>
+        public void Show(ScreenKeyboardShowParams showParams, ScreenKeyboardCallbacks callbacks = default)
```

Comment:
You can do "= default" for both parameters, then there is no reason for Show() with no arguments, since both do the same.